build_tooling/parallel_test.sh

```diff
@@ -2,32 +2,37 @@
 set -e
 
 tooling_dir="$(dirname $BASH_SOURCE)"
-echo Saving results to ${test_output:="$(realpath "$tooling_dir/../cpp/out")"}
-[[ -d "$test_output" ]] || mkdir -p "$test_output"
+echo Saving results to ${TEST_OUTPUT_DIR:="$(realpath "$tooling_dir/../cpp/out")"}
+[[ -d "$TEST_OUTPUT_DIR" ]] || mkdir -p "$TEST_OUTPUT_DIR"
+
+[[ -e ${PARALLEL_TEST_ROOT:=/tmp/parallel_test} ]] && rm -rf $PARALLEL_TEST_ROOT
+splits=${TEST_PARALLELISM:-${CMAKE_BUILD_PARALLEL_LEVEL:-`nproc || echo 2`}}
+
+catch=`{ which catchsegv 2>/dev/null || echo ; } | tail -n 1`
 
 function worker() {
     group=${1:?Must pass the group id argument}
     shift
-    duration_file="$test_output/pytest-durations.$group.json"
-    new_root=/tmp/just_tests$group
+    duration_file="$TEST_OUTPUT_DIR/pytest-durations.$group.json"
+    new_root=$PARALLEL_TEST_ROOT/$group
     set -o xtrace -o pipefail
 
     cp "$tooling_dir/pytest-durations.json" "$duration_file"
 
     # Build a directory that's just the test assets, so can't access other Python source not in the wheel
     # Each test also get a separate directory since there's a mystery lock somewhere preventing concurrent runs (even)
     # from different Python processes
-    mkdir $new_root # Will fail the function if it exists for some reason
-    ln -s "$(realpath "$tooling_dir/../python/tests")" $new_root
+    mkdir -p $new_root
+    MSYS=winsymlinks:nativestrict ln -s "$(realpath "$tooling_dir/../python/tests")" $new_root/
     cd $new_root
 
-    python -m pytest -v --show-capture=no --log-file="$test_output/pytest-logger.$group.log" \
-        --junitxml="$test_output/pytest.$group.xml" \
-        --splits 2 --group $group --durations-path="$duration_file" --store-durations \
-        "$@" 2>&1 | sed -r "s#^(tests/.*/([^/]+\.py))?#$group: \2#"
+    $catch python -m pytest -v --show-capture=no --log-file="$TEST_OUTPUT_DIR/pytest-logger.$group.log" \
+        --junitxml="$TEST_OUTPUT_DIR/pytest.$group.xml" \
+        --splits $splits --group $group --durations-path="$duration_file" --store-durations \
+        "$@" 2>&1 | sed -ur "s#^(tests/.*/([^/]+\.py))?#$group: \2#"
```

Comment:
Crucial `sed` optimization ;)

Reply:
Yeah, vital to Windows as the build used to die without printing out a single output.